### PR TITLE
Run migration to Nx and angular 14 and Storybook 6

### PR DESCRIPTION
**Pull request by fmontes**
### Proposed Changes
* change 1
* change 2

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration/dot-apps-configuration.component.spec.ts [hunk 1]
```diff
@@ -218,10 +216,13 @@ describe('DotAppsConfigurationComponent', () => {
             );
         });
 
-        it('should have dot-apps-configuration-list with correct values', () => {
+        xit('should have dot-apps-configuration-list with correct values', () => {
+            console.log('listComp--> ', 'here');
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration/dot-apps-configuration.component.spec.ts:220]: remove.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration/dot-apps-configuration.component.spec.ts [hunk 2]
```diff
@@ -218,10 +216,13 @@ describe('DotAppsConfigurationComponent', () => {
...
             const listComp = fixture.debugElement.query(
                 By.css('dot-apps-configuration-list')
             ).componentInstance;
+            fixture.detectChanges();
+            console.log('listComp--> ', listComp);
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration/dot-apps-configuration.component.spec.ts:225]: remove.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.spec.ts [hunk 3]
```diff
@@ -138,6 +141,7 @@ describe('DotPortletDetailComponent', () => {
     it('should have dot-contentlets', () => {
         spyOnProperty(router, 'parent', 'get').and.returnValue({
             parent: {
+                // @ts-ignore
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.spec.ts:144]: We leave `lint` out in the tests, something in the migration might be remove that setting, we need it back.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ export class DotPortletDetailComponent implements OnInit {
     constructor(private route: ActivatedRoute) {}
 
     ngOnInit() {
+        // TODO: Change to use this.route.snapshot.paramMap.get('id')
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.ts:15]: Why? can you do it here?

File: core-web/apps/dotcms-ui/src/app/view/components/_common/dot-site-selector/dot-site-selector.component.spec.ts [hunk 5]
```diff
@@ -150,10 +150,6 @@ describe('SiteSelectorComponent', () => {
         expect(comp.handleSitesRefresh).toHaveBeenCalledTimes(1);
     });
 
-    describe('Pagination', () => {
-        //
-    });
-
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/_common/dot-site-selector/dot-site-selector.component.spec.ts:156]: Good catch.

File: core-web/apps/dotcms-block-editor/src/test.ts [hunk 6]
```diff
@@ -7,6 +7,7 @@ import {
     platformBrowserDynamicTesting
 } from '@angular/platform-browser-dynamic/testing';
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (fmontes) [inline core-web/apps/dotcms-block-editor/src/test.ts:10]: Who's crying about this? lint, typescript or vscoe?

File: core-web/apps/dotcms-ui/.eslintrc.json [hunk 7]
```diff
@@ -24,7 +24,8 @@
                         "prefix": "dot",
                         "style": "kebab-case"
                     }
-                ]
+                ],
+                "@typescript-eslint/ban-ts-comment": ["warn"]
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/.eslintrc.json:28]: This doesn't look safe to me. We don't want allow people to disable types.

File: core-web/apps/dotcms-ui/.storybook/main.js [hunk 8]
```diff
@@ -11,6 +11,9 @@ module.exports = {
         '../src/**/*.stories.@(js|jsx|ts|tsx)',
     ],
     addons: [...rootMain.addons],
+    features: {
+        previewMdx2: true,
+    },
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/.storybook/main.js:16]: What is this for?

File: core-web/libs/block-editor/.storybook/main.js [hunk 9]
```diff
@@ -1,11 +1,29 @@
...
+      config = await rootMain.webpackFinal(config, { configType });
+    }
+
+
+
+    // add your own webpack tweaks if needed
+
+    return config;
+  },
+};
```
Comment (fmontes) [inline core-web/libs/block-editor/.storybook/main.js:29]: I'm guessing this is the new way to add custom config to webpack in Storybook 6.5?

File: core-web/libs/dotcms-webcomponents/.storybook/tsconfig.json [hunk 10]
```diff
@@ -1,8 +1,7 @@
-{
-  "extends": "../tsconfig.base.json",
```
Comment (fmontes) [inline core-web/libs/dotcms-webcomponents/.storybook/tsconfig.json:2]: Why not extends anymore?

File: core-web/package.json [hunk 11]
```diff
@@ -37,16 +37,13 @@
     "format": "nx format:write",
     "format:write": "nx format:write",
     "format:check": "nx format:check",
-    "update": "nx migrate latest",
-    "workspace-generator": "nx workspace-generator",
     "dep-graph": "nx dep-graph",
-    "help": "nx help",
     "pretty-quick": "pretty-quick",
-    "prepare": "is-ci || cd ../ && pwd && husky install core-web/.husky"
+    "prepare": "cd .. && node core-web/prepare.js"
```
Comment (fmontes) [inline core-web/package.json:42]: This change was introduce by Victor to fix the CICD, are you sure this won't break it again?

File: core-web/package.json [hunk 12]
```diff
@@ -37,16 +37,13 @@
     "format": "nx format:write",
     "format:write": "nx format:write",
     "format:check": "nx format:check",
-    "update": "nx migrate latest",
-    "workspace-generator": "nx workspace-generator",
```
Comment (fmontes) [inline core-web/package.json]: Why?

File: core-web/package.json [hunk 13]
```diff
@@ -37,16 +37,13 @@
...
     "dep-graph": "nx dep-graph",
-    "help": "nx help",
```
Comment (fmontes) [inline core-web/package.json]: Why?

File: core-web/package.json (hunk 11 above)
Comment (oidacra) [reply core-web/package.json]: I sent the change to him.

File: core-web/package.json [hunk 14]
```diff
@@ -116,6 +113,7 @@
     "@babel/core": "^7.12.9",
     "@compodoc/compodoc": "^1.1.11",
     "@materia-ui/ngx-monaco-editor": "^6.0.0",
+    "@mdx-js/react": "^2.1.2",
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/package.json:116]: its related to make run storybook in dotcms-ui
features: { previewMdx2: true, },

File: core-web/package.json [hunk 15]
```diff
@@ -136,6 +134,7 @@
     "@storybook/builder-webpack5": "6.5.9",
     "@storybook/html": "^6.5.9",
     "@storybook/manager-webpack5": "6.5.9",
+    "@storybook/mdx2-csf": "^0.0.3",
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/package.json:137]: its related to make run storybook in dotcms-ui 
`features: {
        previewMdx2: true,
    },`

File: core-web/apps/dotcdn/project.json [hunk 16]
```diff
@@ -1,6 +1,6 @@
 {
+    "$schema": "../../node_modules/nx/schemas/project-schema.json",
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcdn/project.json:2]: that's come from the migration automatically.

File: core-web/libs/block-editor/jest.config.ts [hunk 17]
```diff
@@ -1,4 +1,5 @@
-module.exports = {
+/* eslint-disable */
```
Comment (rjvelazco) [inline core-web/libs/block-editor/jest.config.ts:1]: Are we getting `eslint` warnings in this file?

File: core-web/libs/dotcms-webcomponents/.storybook/main.js [hunk 18]
```diff
@@ -0,0 +1,29 @@
+const rootMain = require('../../../.storybook/main');
+
+
+
+module.exports = {
```
Comment (rjvelazco) [inline core-web/libs/dotcms-webcomponents/.storybook/main.js:5]: Can we add a comment with a link to the documentation where we found this configuration? Something like:

// Find the configuration here: https://storybook.js.org/

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 19]
```diff
@@ -98,18 +98,15 @@ jobs:
...
           cd core-web
           npm install
-      - name:
-          Building for production
-          ## Build dotcms-webcomponents because dotcms-ui doesn't have the lib as a direct dep
-        ## so Nx can't be identified and build it.
+      - name: Building for production
         run: |
           cd core-web
-          npm run nx build dotcms-webcomponents && npm run nx build dotcms-ui -- --prod
```
[the remark this answers is not in the document]
Comment (oidacra) [reply .github/workflows/main.yml:107]: I configure the project dotcms-ui to depend on dotcms-webcomponent. Nx does that task automatically.

File: core-web/apps/dotcdn-e2e/project.json [hunk 20]
```diff
@@ -1,5 +1,5 @@
 {
-    "root": "apps/dotcdn-e2e",
+    "$schema": "../../node_modules/nx/schemas/project-schema.json",
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcdn-e2e/project.json:2]: With the migration of Nx, automatically update that. It's the schema for the file.

File: core-web/migrations.json [hunk 21]
```diff
@@ -0,0 +1,264 @@
+{
+  "migrations": [
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/migrations.json]: removed

File: core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration/dot-apps-configuration.component.spec.ts (hunk 2 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-apps/dot-apps-configuration/dot-apps-configuration.component.spec.ts]: Removed.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.ts (hunk 4 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.ts]: Removed, it was left as it was.

File: core-web/apps/dotcms-ui/.storybook/main.js (hunk 8 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/.storybook/main.js:16]: Is to force use "@mdx-js/react": "^2.1.2 in storybook

File: core-web/package.json (hunk 15 above)
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/package.json:139]: and "@mdx-js/react": "^2.1.2

File: core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.spec.ts (hunk 3 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-porlet-detail/dot-portlet-detail.component.spec.ts]: removed

File: core-web/apps/dotcms-block-editor/src/test.ts (hunk 6 above)
Comment (oidacra) [reply core-web/apps/dotcms-block-editor/src/test.ts:10]: Removed. The editor, lint, build and test works fine.

File: core-web/libs/dotcms-webcomponents/.storybook/tsconfig.json (hunk 10 above)
Comment (oidacra) [reply core-web/libs/dotcms-webcomponents/.storybook/tsconfig.json:2]: I put back the extends.

File: core-web/apps/dotcms-ui/.eslintrc.json [hunk 22]
```diff
@@ -1,13 +1,14 @@
 {
     "extends": ["../../.eslintrc.json"],
-    "ignorePatterns": ["!**/*"],
+    "ignorePatterns": ["!**/*", "*.spec.ts"],
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/.eslintrc.json:3]: removed.

File: core-web/tsconfig.base.json [hunk 23]
```diff
@@ -8,9 +8,9 @@
         "emitDecoratorMetadata": true,
         "experimentalDecorators": true,
         "importHelpers": true,
-        "target": "es2015",
+        "target": "es2020",
         "module": "esnext",
-        "lib": ["es2017", "dom", "dom.iterable"],
+        "lib": ["es2020", "dom", "dom.iterable"],
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/tsconfig.base.json:13]: changed to es2015 and es2017 in libs.

File: core-web/package.json [hunk 24]
```diff
@@ -20,8 +20,9 @@
     "build:docs:dotcms-models": "typedoc --options libs/dotcms-models/typedoc.json",
     "publish:docs": "gh-pages --dist ./docs --dest ./",
     "build:dev": "nx build dotcms-webcomponents --with-deps;nx build dotcms-ui",
-    "build:prod": "npx nx run dotcms-ui:build:production",
+    "build:prod": "npm run build:dotcms-webcomponents && npx nx run dotcms-ui:build:production",
     "build:dotcms-block-editor": "ng build dotcms-block-editor --configuration production --optimization --output-hashing=none && cat dist/apps/dotcms-block-editor/{generator-runtime,runtime,polyfills,main}.js > ../dotCMS/src/main/webapp/html/dotcms-block-editor.js",
+    "build:dotcms-webcomponents": "npx nx build dotcms-webcomponents",
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/package.json:25]: npx removed. In the end not used. With the explicit dependency of the lib in `dotcms-ui`, it's not necessary.

File: core-web/apps/dotcms-ui/.eslintrc.json [hunk 25]
```diff
@@ -24,8 +25,7 @@
                         "prefix": "dot",
                         "style": "kebab-case"
                     }
-                ],
-                "@typescript-eslint/ban-ts-comment": ["warn"]
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/.eslintrc.json]: removed.

File: core-web/apps/dotcms-ui/project.json [hunk 26]
```diff
@@ -108,6 +109,7 @@
         },
         "test": {
             "executor": "@angular-devkit/build-angular:karma",
+            "dependsOn": ["build"],
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/project.json]: removed. Not necessary.

File: core-web/apps/dotcms-ui/project.json [hunk 27]
```diff
@@ -79,7 +79,8 @@
                     ]
                 }
             },
-            "defaultConfiguration": ""
+            "defaultConfiguration": "",
+            "dependsOn": ["^build"]
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/project.json:83]: Necessary to the explicit dependency of` dotcms-webcomponent` need `build` that dependency first.

[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/project.json:83]: https://nx.dev/configuration/projectjson

File: core-web/apps/dotcms-ui/src/app/portlets/shared/dot-content-types-edit/components/fields/content-type-fields-properties-form/field-properties/name-property/name-property.component.ts [hunk 28]
```diff
@@ -1,5 +1,5 @@
 import { Component } from '@angular/core';
-import { FormGroup } from '@angular/forms/forms';
+import { FormGroup } from '@angular/forms';
```
[the remark this answers is not in the document]
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/shared/dot-content-types-edit/components/fields/content-type-fields-properties-form/field-properties/name-property/name-property.component.ts]: Replaced.

**Review by hmoreras — APPROVED** [review]

**Pull request merged**